.github/actions/terraform_state_artifact/action.yml

```diff
@@ -0,0 +1,202 @@
+name: 'terraform_state_artifact'
+description: 'Sets up and runs Terraform, and creates an encrypted Terraform artifact'
+author: 'Sturla Bragason'
+inputs:
+  encryptionkey:
+    description: 'Used to read artifact and as a key to encrypt and decrypt the state file artifact'
+    required: true
+  apply:
+    description: 'terraform apply'
+    required: false
+    default: true
+  custom_plan_flags:
+    description: 'Add custom flags to the terraform plan command'
+    required: false
+    default: ''
+  custom_apply_flags:
+    description: 'Add custom flags to the terraform apply command'
+    required: false
+    default: ''
+  path:
+    description: 'Terraform configuration path'
+    required: false
+    default: ''
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v2
+    - uses: hashicorp/setup-terraform@v1
+    - id: terraform
+      working-directory: ${{ inputs.path }}
+      run: |
+
+        $Repo = "${{ github.repository }}"
+        $BaseUri = "https://api.github.com"
+        $ArtifactUri = "$BaseUri/repos/$Repo/actions/artifacts"
+        $Token = "${{ github.token }}" | ConvertTo-SecureString -AsPlainText
+        $RestResponse = Invoke-RestMethod -Authentication Bearer -Uri $ArtifactUri -Token $Token | Select-Object -ExpandProperty artifacts
+        if ($RestResponse){
+          $MostRecentArtifactURI = $RestResponse | Sort-Object -Property created_at -Descending | where name -eq "terraformstatefile" | Select-Object -First 1 | Select-Object -ExpandProperty archive_download_url
+          Write-Host "Most recent artifact URI = $MostRecentArtifactURI"
+          if ($MostRecentArtifactURI){
+            Invoke-RestMethod -uri $MostRecentArtifactURI -Token $Token -Authentication bearer -outfile ./state.zip
+            Expand-Archive ./state.zip
+            openssl enc -d -in ./state/terraform.tfstate.enc -aes-256-cbc -pbkdf2 -pass pass:"${{ inputs.encryptionkey }}" -out ./terraform.tfstate
+          }
+        }
+
+        $tfdir = "${{ inputs.path }}"
+
+        npm install -g twilio-cli
+
+        terraform init
+        
+        #############################################################################
+        ### THE FOLLOWING CAPTURES THE SIDS FROM THE ENVIRONMENT AND IMPORTS THEM
+        ### INTO TERRAFORM IF THEY EXIST 
+        #############################################################################
```

Comment:
IMO this script adds some tech debt that we should try to rectify:

- API calls are repeated, even though the result can be stored and re-parsed
- It would be better if there were structured files defining the objects being fetched and mapped. For example, a json that maps activity name to the terraform input it goes to, same with serverless, studio, etc.